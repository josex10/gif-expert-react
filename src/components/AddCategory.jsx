import { useState } from "react";

export const AddCategory = () => {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = ({ target }) => setInputValue(target.value);
  const handleSubmit = (event) => {
    event.preventDefault();
  };
  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Search Gifts"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};
