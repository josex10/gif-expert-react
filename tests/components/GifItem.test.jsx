import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Testing component GifItem", () => {
  const image = {
    title: "Messi",
    url: "https://my-messi.gif/",
  };

  test("Should be a match with the snapshot", () => {
    const { container } = render(<GifItem {...image} />);
    expect(container).toMatchSnapshot();
  });

  test("Should be show the image  with the correct URL and ALT", () => {
    render(<GifItem {...image} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(image.url);
    expect(alt).toBe(image.title);
  });

  test("Should b show the right title", () => {
    render(<GifItem {...image} />);
    expect(screen.getByText(image.title)).toBeTruthy();
  });
});
