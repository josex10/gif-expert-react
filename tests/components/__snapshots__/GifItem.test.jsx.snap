// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component GifItem Should be a match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Messi"
      src="https://my-messi.gif/"
    />
    <p>
      Messi
    </p>
  </div>
</div>
`;
